frontends/election-manager/src/components/print_all_ballots_button.tsx

```diff
@@ -0,0 +1,362 @@
+import pluralize from 'pluralize';
+import {
+  getBallotStylesData,
+  sortBallotStyleDataByPrecinct,
+} from '../utils/election';
+
+import { AppContext } from '../contexts/app_context';
+import { HandMarkedPaperBallot } from './hand_marked_paper_ballot';
+import { Button } from './button';
+import { LinkButton } from './link_button';
+
+import {
+  BallotMode,
+  ballotModeToReadableString,
+  PrintableBallotType,
+} from '../config/types';
+import { BallotTypeToggle } from './ballot_type_toggle';
+import { BallotModeToggle } from './ballot_mode_toggle';
+import { PrintableArea } from './printable_area';
+import { DEFAULT_LOCALE } from '../config/globals';
+import { BallotCopiesInput } from './ballot_copies_input';
+import { PrintBallotButtonText } from './print_ballot_button_text';
+
+export const PRINTER_WARMUP_TIME = 8300;
+export const TWO_SIDED_PRINT_TIME = 3300;
+export const PRINTER_COOLDOWN_TIME = 5000;
+
+const CenteredOptions = styled.div`
+  text-align: center;
+  & > :last-child {
+    margin-bottom: 0;
+  }
+`;
+
+type ModalState = 'no-printer' | 'options' | 'printing';
+
+const defaultBallotLocales: BallotLocales = { primary: DEFAULT_LOCALE };
+
+interface Props {
+  draftMode?: boolean;
+}
+
+export function PrintAllBallotsButton({ draftMode }: Props): JSX.Element {
+  const makeCancelable = useCancelablePromise();
+  const {
+    electionDefinition,
+    auth,
+    logger,
+    hasPrinterAttached,
+    printer,
+    addPrintedBallot,
+  } = useContext(AppContext);
+  assert(electionDefinition);
+  assert(isAdminAuth(auth) || isSuperadminAuth(auth));
+  const userRole = auth.user.role;
+  const { election, electionHash } = electionDefinition;
+
+  const needsPrinter = window.kiosk && !hasPrinterAttached;
+  const initialState: ModalState = needsPrinter ? 'no-printer' : 'options';
+  const printLock = useLock();
+
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [modalState, setModalState] = useState<ModalState>(initialState);
+  const [printFailed, setPrintFailed] = useState(false);
+
+  const [ballotMode, setBallotMode] = useState(
+    draftMode ? BallotMode.Draft : BallotMode.Official
+  );
+  const [isAbsentee, setIsAbsentee] = useState(true);
+  const [ballotCopies, setBallotCopies] = useState(1);
+
+  const [ballotIndex, setBallotIndex] = useState<number>();
+  const ballotStyles = useMemo<BallotStyleData[]>(
+    () =>
+      sortBallotStyleDataByPrecinct(election, getBallotStylesData(election)),
+    [election]
+  );
+
+  useEffect(() => {
+    if (draftMode) {
+      setBallotMode(BallotMode.Draft);
+    } else {
+      setBallotMode(BallotMode.Official);
+    }
+  }, [draftMode]);
+
+  useEffect(() => {
+    if (hasPrinterAttached) {
+      if (modalState === 'no-printer') {
+        setModalState('options');
+      }
+    } else if (!hasPrinterAttached) {
+      if (modalState === 'options') {
+        setModalState('no-printer');
+      }
+    }
```

Comment:
This might be a bit easier to parse if we collapse the nested if statements, e.g.

```ts
if (hasPrinterAttached && modalState === 'no-printer') {
  ...
} else if (!hasPrinterAttached && modalState === 'options') {
  ...
}
```